code/modules/martial_arts/cqc.dm

```diff
@@ -10,12 +11,42 @@
 /datum/martial_art/cqc/under_siege
 	name = "Close Quarters Cooking"
 
+/datum/martial_art/cqc/under_siege/teach(mob/living/carbon/human/H, make_temporary)
+	RegisterSignal(H, COMSIG_AREA_ENTERED, PROC_REF(kitchen_check))
+	return ..()
+
+/datum/martial_art/cqc/under_siege/remove(mob/living/carbon/human/H)
+	UnregisterSignal(H, COMSIG_AREA_ENTERED)
+	. = ..()
+
+/datum/martial_art/cqc/under_siege/proc/kitchen_check(mob/living/carbon/human/H, area/entered_area)
+	SIGNAL_HANDLER //COMSIG_AREA_ENTERED
+	var/list/held_items = list(H.get_active_hand(), H.get_inactive_hand())
+	if(!(is_type_in_typecache(entered_area, areas_under_siege)))
+		for(var/obj/item in held_items)
+			if(istype(item, /obj/item/slapper/parry))
+				var/obj/item/slapper/parry/smacking_hand = item
+				qdel(smacking_hand)
```

Comment:
You probably don't need to do the extra cast here, you can save yourself a bit of time here I think
```suggestion
		for(var/obj/item/slapper/parry/smacking_hand in held_items)
				qdel(smacking_hand)
```